partitura/musicanalysis/performance_features.py

```diff
@@ -0,0 +1,488 @@
+
+
+def compute_matched_score(score: ScoreLike, 
+                            performance: PerformanceLike,
+                            alignment: list,):
+    """
+    Compute the matched score and add the score features
+
+    Parameters
+    ----------
+    score : partitura.score.ScoreLike
+        Score information, can be a part, score
+    performance : partitura.performance.PerformanceLike
+        Performance information, can be a ppart, performance
+    alignment : list
+        The score--performance alignment, a list of dictionaries
+
+    Returns
+    -------
+    m_score : np strutured array
+    unique_onset_idxs : list
+    """
+
+    m_score, snote_ids = to_matched_score(score, performance, alignment, include_score_markings=True)
+    (time_params, unique_onset_idxs) = encode_tempo(
+        score_onsets=m_score["onset"],
+        performed_onsets=m_score["p_onset"],
+        score_durations=m_score["duration"],
+        performed_durations=m_score["p_duration"],
+        return_u_onset_idx=True,
+        tempo_smooth="average"
+    )
+    m_score = rfn.append_fields(m_score, "beat_period", time_params['beat_period'], "f4", usemask=False)
+    m_score = rfn.append_fields(m_score, "id", snote_ids, "U256", usemask=False)
+    return m_score, unique_onset_idxs, snote_ids
+
+
+def list_performance_feats_functions():
+    """Return a list of all feature function names defined in this module.
+
+    The feature function names listed here can be specified by name in
+    the `compute_performance_features` function. For example:
+
+    >>> feature, names = compute_performance_features(score,
+                                                    performance,
+                                                    alignment,
+                                                    ['asynchrony_feature'])
+
+    Returns
+    -------
+    list
+        A list of strings
+
+    """
+    module = sys.modules[__name__]
+    bfs = []
+    exclude = {"make_feature"}
+    for name in dir(module):
+        if name in exclude:
+            continue
+        member = getattr(sys.modules[__name__], name)
+        if isinstance(member, types.FunctionType) and name.endswith("_feature"):
+            bfs.append(name)
+    return bfs
+
+
+def print_performance_feats_functions():
+    """Print a list of all feature function names defined in this module,
+    with descriptions where available.
+
+    """
+    module = sys.modules[__name__]
+    doc_indent = 4
+    for name in list_performance_feats_functions():
+        print("* {}".format(name))
+        member = getattr(sys.modules[__name__], name)
+        if member.__doc__:
+            print(
+                " " * doc_indent + member.__doc__.replace("\n", " " * doc_indent + "\n")
+            )
+
+
+
+# alias
+list_performance_feature_functions = list_performance_feats_functions
+print_performance_feature_functions = print_performance_feats_functions
+
+
+def map_fields(note_info, fields):
```

Comment:
I don't fully understand what this function is supposed to do.

Reply:
I don't know either, but it's unused for the the performance features in this PR, I just forgot to delete it. Addressed in commit [cc7476a](https://github.com/CPJKU/partitura/pull/260/commits/cc7476adf9b3f1abb9a2f7779b3e694c251119f8)